File: Programmers/Segun/3주차(Stack, Queue)/같은 숫자는 싫어.cpp
//스택의 개념을 사용해서 전에 들어왔던 값이랑 비교해서 다른값이 들어왔을때만 vector에 넣어주게 작성

#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> arr) 
{
    vector<int> answer; //리턴해주는 결과값
    for(int i=0;i<arr.size();i++){  //주어진 배열 탐색
        if(arr[i]!=arr[i-1]||i==0){answer.push_back(arr[i]);}   //다른값이 들어왔으면 넣어주기
    }
    
    return answer;
}

File: Programmers/BoHwa/3주차(Stack, Queue)/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        int[] answer;
        ArrayList<Integer> arrList = new ArrayList<>(); //배열에 계속 추가하기 위해 ArrayList를 사용
        int num=-1;

        for (int i=0; i<arr.length; i++){
            if (num!=arr[i]){ //마지막에 저장된 값과 현재 arr의 값이 같지 않으면 저장하고 num에 현재 숫자 저장
                arrList.add(arr[i]);
                num=arr[i];
            }
        }

        answer=new int[arrList.size()]; //int형 배열에 ArrayList 값을 넣기
        for (int i=0; i<arrList.size(); i++){
            answer[i]=arrList.get(i);
        }

        return answer;
    }
}
